core/src/main/scala/org/apache/spark/scheduler/dynalloc/ExecutorMonitor.scala

```diff
@@ -52,6 +53,7 @@ private[spark] class ExecutorMonitor(
     conf.get(DYN_ALLOCATION_SHUFFLE_TRACKING)
 
   private val executors = new ConcurrentHashMap[String, Tracker]()
+  private val execResourceProfileCount = new mutable.HashMap[Int, Int]()
```

Comment:
Its a bit surprising that this doesn't need to be a CHM, when `executors` does (because its referenced from `ExecutorAllocationManager.schedule()` method).  I think it looks OK on a first pass just want to check you'd checked this as well.  I guess the key thing is that `timedOutExecutors()` doesn't actually need to know anything about ResourceProfiles.

Reply:
I actually changed this to be CHM in the latest code, I will update the pr